fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/FieldValue.java

```diff
@@ -174,6 +176,14 @@ private static Object unwrapPrimitive(@Nonnull Type type, @Nullable Object field
             return returnList;
         } else if (type.getTypeCode() == Type.TypeCode.VERSION) {
             return FDBRecordVersion.fromBytes(((ByteString)fieldValue).toByteArray(), false);
+        } else if (type.isUuid()) {
+            if (fieldValue instanceof UUID) {
+                return fieldValue;
+            }
+            Verify.verify(fieldValue instanceof DynamicMessage);
+            final var message = (DynamicMessage) fieldValue;
+            return new UUID((Long) message.getField(message.getDescriptorForType().findFieldByName("most_significant_bits")),
```

Comment:
Can we cache this? The lookup is a binary search iirc.

Reply:
Caching aside, but should the descriptor not be one of our predefined external dependencies when the dynamic types are defined (as per this PR)? If that is the case the field is also always the same and can be stashed somewhere. If none of that is true (I would like to understand where the reasoning is wrong), I still do not like the literal use of `most_significant_bits`. We normally treat protobuf in a way that we can rename things without any problem. This line of code make that assumption break (even using an index is better here). You can also use the statically defined proto-class' field name, I suppose.